docs/blog/2019-04-01-setting-up-yarn-workspaces-for-theme-development/index.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Setting up Yarn Workspaces for Theme Development
+date: 2019-04-01
+author: Brent Jackson
+excerpt: ""
+tags:
+  - themes
+  - yarn
+  - tutorials
+---
+
+Gatsby [themes][themes-tag] are a new way to share functionality across multiple Gatsby sites.
+Using [Yarn workspaces][] is an excellent way to set up a project for theme development.
+It allows you to keep multiple packages in a single parent directory and link dependencies together.
+For Gatsby theme development, that means you can keep multiple themes and example sites together in a single project.
+In this post, we'll walk through how to use Yarn workspaces to set up a development environment for creating custom themes.
+
+## Installation and directory structure
+
+First, if you don't have Yarn installed already, follow the directions here to [install Yarn][].
+Next, create a new directory for the entire project, where you'll be adding your theme and an example site later.
+
+Create two subdirectories in this folder: one for the theme itself and one for the example site.
+
+```shell
+mkdir gatsby-theme-example-workspaces example
+```
+
+Add a `package.json` to the root directory with these subdirectories in the `workspaces` field. Note that setting `"private": true` is required for Yarn workspaces.
+
+```json:title=package.json
+{
+  "private": true,
+  "workspaces": ["gatsby-theme-example-workspaces", "example"]
+}
+```
+
+Switch to each subdirectory and run `yarn init -y` to create a `package.json` for each one.
+Be sure the `name` field in your theme's `package.json` matches the directory name exactly.
+This is currently a limitation of the shadowing feature, not Yarn workspaces.
```

Comment:
pls explain also: which shadowing feature?

Reply:
It could say `This is currently a limitation of Gatsby theme shadowing` instead?